### PR TITLE
Removed the hard coded strings

### LINKED ISSUE #430: Removal of hard coded strings
**Summary**
The hard coded strings should be removed.

**Device and android version**
Device: Vivo V11 pro
Android version: 8.1.0

### PR DESCRIPTION
**Pull request by gopikaharikumar**
Fixes #430

### CODE REVIEW

**Comment by gopikaharikumar** [comment]
@cpg can you please review this PR?

File: src/main/java/org/amahi/anywhere/activity/OfflineFilesActivity.java [hunk 1]
```diff
@@ -168,7 +168,7 @@ public void onFileCopied(FileCopiedEvent event) {
     }
 
     private void dismissPreparingDialog() {
-        PrepareDialogFragment fragment = (PrepareDialogFragment) getSupportFragmentManager().findFragmentByTag("prepare_dialog");
+        PrepareDialogFragment fragment = (PrepareDialogFragment) getSupportFragmentManager().findFragmentByTag(getString(R.string.prepare_dialog));
```
Comment (codedentwickler) [inline src/main/java/org/amahi/anywhere/activity/OfflineFilesActivity.java]: I dont see any reason for adding this to string.xml. It is not really a string resource, it is more of an id. We wouldn't want this to change if we decide to translate the App strings later. A better way would be to move it to the Constants class.

File: src/main/java/org/amahi/anywhere/activity/RecentFilesActivity.java [hunk 2]
```diff
@@ -117,7 +117,7 @@ private void showIntroductoryOverlay() {
             new Handler().post(() -> {
                 mIntroductoryOverlay = new IntroductoryOverlay
                     .Builder(this, mediaRouteMenuItem)
-                    .setTitleText("Introducing Cast")
+                    .setTitleText(R.string.introducing_cast)
```
Comment (codedentwickler) [inline src/main/java/org/amahi/anywhere/activity/RecentFilesActivity.java]: Awesome.

File: src/main/java/org/amahi/anywhere/activity/RecentFilesActivity.java [hunk 3]
```diff
@@ -394,7 +394,7 @@ private void showFileDownloadingFragment(ServerFile file) {
     }
 
     private void dismissPreparingDialog() {
-        PrepareDialogFragment fragment = (PrepareDialogFragment) getSupportFragmentManager().findFragmentByTag("prepare_dialog");
+        PrepareDialogFragment fragment = (PrepareDialogFragment) getSupportFragmentManager().findFragmentByTag(getString(R.string.preparedialog));
```
Comment (codedentwickler) [inline src/main/java/org/amahi/anywhere/activity/RecentFilesActivity.java]: Same here, better in the constant class.

File: src/main/java/org/amahi/anywhere/activity/ServerFilesActivity.java [hunk 4]
```diff
@@ -389,7 +389,7 @@ private void openCamera() {
 
     private File createImageFile() throws IOException {
         String timeStamp = String.valueOf(new Date().getTime());
-        String imageFileName = "photo-" + timeStamp;
+        String imageFileName = getString(R.string.photo_) + timeStamp;
```
Comment (codedentwickler) [inline src/main/java/org/amahi/anywhere/activity/ServerFilesActivity.java]: Same here.

File: src/main/java/org/amahi/anywhere/fragment/SettingsFragment.java [hunk 5]
```diff
@@ -112,7 +112,7 @@ private String getApplicationVersionSummary() {
     }
 
     private String getAutoUploadSummary() {
-        return isUploadEnabled() ? "Enabled" : "Disabled";
+        return isUploadEnabled() ? getString(R.string.enabled) : getString(R.string.disabled);
```
Comment (codedentwickler) [inline src/main/java/org/amahi/anywhere/fragment/SettingsFragment.java]: Nice

File: src/main/java/org/amahi/anywhere/service/UploadService.java [hunk 6]
```diff
@@ -219,7 +219,7 @@ private void setUpUploadManager() {
 
     private String queryImagePath(Uri imageUri) {
         String filePath = null;
-        if ("content".equals(imageUri.getScheme())) {
+        if (getString(R.string.content).equals(imageUri.getScheme())) {
```
Comment (codedentwickler) [inline src/main/java/org/amahi/anywhere/service/UploadService.java]: Same here.

**Review by codedentwickler — CHANGES_REQUESTED** [review]
Good work overall. Move strings that should always remain the same (i.e ids or tags)to a Constants class. If this class do not exists, you can create one.

**Comment by cpg** [comment]
@gopikaharikumar please take a look at the recommendations and see to either merge them or suggest what to do about it.

File: src/main/res/values/strings.xml [hunk 7]
```diff
@@ -169,10 +169,35 @@
...
-
+    <string name="connection">CONNECTION</string>
+    <string name="shares">shares</string>
+    <string name="unknown_artist">Unknown Artist</string>
+    <string name="android_amahi_anywhere">Android Amahi Anywhere</string>
+    <string name="image_mediatype">image/*</string>
+    <string name="md5">MD5</string>
+    <string name="prepare_dialog_2">prepare_dialog</string>
+    <string name="progressdialog">progress_dialog</string>
+    <string name="prepare_dialog_3">prepare_dialog</string>
```
Comment (cpg) [inline src/main/res/values/strings.xml:182]: Is this and others (identical!) strings that the users see?

What should it be @octacode?

Comment (codedentwickler) [reply src/main/res/values/strings.xml:182]: They are not seen by the user. Putting them in strings.xml means they would be translated if we choose to localize the Amahi app in the future.

**Comment by codedentwickler** [comment]
@gopikaharikumar As I said earlier, moving string constants the user do not see to strings.xml is not appropriate. Moving them to a Constants class where it remains the same even if we translate later is safer. 

This PR fixes a minor issue, please make the changes so we can merge it.

**Comment by sumit-kushwah** [comment]
Can I work on this ?

**Comment by chirag-jn** [comment]
Instead of adding constant strings to a Constants class, we can also add them to strings.xml and keep an attribute `translable="false"` in strings.xml itself.

**Comment by cpg** [comment]
Fixed by #539. Good work!

**Pull request closed without merging**